src/Common/src/System/Text/ValueStringBuilder.cs

```diff
@@ -0,0 +1,189 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
```

Comment:
is this worth tests of its own?

Reply:
especially as the growth path is not getting exercised, unless tests need more than 128 hex chars?

Reply:
> unless tests need more than 128 hex chars?

There are tests that need more than 128 hex chars.

> is this worth tests of its own?

I can add some.